components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/impl/ActionInvocationResponseClassFactory.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.action.execution.impl;
+
+import org.wso2.carbon.identity.action.execution.ActionInvocationResponseClassProvider;
+import org.wso2.carbon.identity.action.execution.DefaultActionInvocationResponseClassProvider;
+import org.wso2.carbon.identity.action.execution.model.ActionType;
+import org.wso2.carbon.identity.action.execution.model.ResponseData;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class defines the classes for extended implementations of action invocation responses for
```

Comment:
This class defines the classes -> This class manages ActionInvocationResponseClassProvider implementations that extends action invocation responses bla bla bla